config/types/etcd.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/coreos/ignition/config/validate/report"
+)
+
+var (
+	InvalidEtcdVersion = errors.New("Etcd version specified is not valid")
+)
+
+// Options can be the options for any Etcd version
+type Options interface{}
+
+type etcdCommon Etcd
+
+type EtcdVersion string
+
+func (e EtcdVersion) Validate() report.Report {
+	validTags := []string{
+		"v3.1.2",
+		"v3.1.1",
+		"v3.1.0",
+		"v3.0.17",
+		"v3.0.16",
+		"v3.0.15",
+		"v3.0.14",
+		"v3.0.13",
+		"v3.0.12",
+		"v3.0.11",
+		"v3.0.10",
+		"v3.0.9",
+		"v3.0.8",
+		"v3.0.7",
+		"v3.0.6",
+		"v3.0.5",
+		"v3.0.4",
+		"v3.0.3",
+		"v3.0.2",
+		"v3.0.1",
+		"v3.0.0",
+		"v2.3.8",
+		"v2.3.7",
+		"v2.3.6",
+		"v2.3.5",
+		"v2.3.4",
+		"v2.3.3",
+		"v2.3.2",
+		"v2.3.1",
+		"v2.3.0",
+	}
+	for _, tag := range validTags {
+		if string(e) == tag {
```

Comment:
Maybe it would be worth running these through the semver library to just check major versions? Otherwise we'll need a new clct release whenever there's a new etcd release.